src/timer_registry.rs

```diff
@@ -1,29 +1,19 @@
 use crate::time_handler_guard::TimeHandlerGuard;
+use crate::timer::{Timer, TimerListener};
+use event_listener::Event;
 use std::collections::{BTreeMap, VecDeque};
 use std::sync::{RwLock, RwLockWriteGuard};
 use std::time::Duration;
-use tokio::sync::{broadcast, oneshot};
-
-type TimersByTime = BTreeMap<Duration, VecDeque<oneshot::Sender<TimeHandlerGuard>>>;
 
+#[derive(Default)]
 pub struct TimerRegistry {
 	current_time: RwLock<Duration>,
-	timers_by_time: RwLock<TimersByTime>,
-	any_timer_scheduled_signal: broadcast::Sender<()>,
-	advance_time_lock: tokio::sync::Mutex<()>,
+	timers_by_time: RwLock<TriggersByTime>,
```

Comment:
`timers_by_time` => `triggers_by_time`?

Reply:
fixed!

Reply:
Nevermind, I just renamed it back. `TriggersByTime` should have been `TimersByTime` instead.